### PR TITLE
contrib: update libbpf to 1.2.2

### PR DESCRIPTION
**Pull request by mmat11**
best reviewed by commit

### CODE REVIEW

File: GPL/Events/Process/Probe.bpf.c [hunk 1]
```diff
@@ -369,9 +384,19 @@ static int tty_write__enter(struct kiocb *iocb, struct iov_iter *from)
...
     }
 
-    u64 nr_segs             = BPF_CORE_READ(from, nr_segs);
-    nr_segs                 = nr_segs > MAX_NR_SEGS ? MAX_NR_SEGS : nr_segs;
-    const struct iovec *iov = BPF_CORE_READ(from, iov);
+    u64 nr_segs = BPF_CORE_READ(from, nr_segs);
+    nr_segs     = nr_segs > MAX_NR_SEGS ? MAX_NR_SEGS : nr_segs;
+
+    const struct iovec *iov;
+    if (bpf_core_field_exists(struct iov_iter____new, __iov)) {
```
Comment (nicholasberlin) [inline GPL/Events/Process/Probe.bpf.c:391]: Should this be:
```c
if (bpf_core_field_exists(struct iov_iter, __iov)) {
```

Seems like we are asking about our local struct definition.

Comment (mmat11) [reply GPL/Events/Process/Probe.bpf.c:391]: I'm debugging this now, it seems like the relocation is not filled regardless. 
Doing
```
if (bpf_core_field_exists(struct iov_iter, __iov)) {
```
fails with a compiler error even though it shouldn't.
I am now trying to fill the offset from userspace

Comment (mmat11) [reply GPL/Events/Process/Probe.bpf.c:391]: Anyway, I added new kernels (all 6.x mainline) to the CI so now it fails properly

**Review by nicholasberlin — APPROVED** [review]
LGTM, if CI passes

**Comment by mmat11** [comment]
> LGTM, if CI passes

There's a broken CI step, but the tests are now passing

**Review by pkoutsovasilis — APPROVED** [review]
LGTM

**Pull request merged**